calliope/backend/checks.py

```diff
@@ -8,14 +8,43 @@
 Checks for model consistency and possible errors when preparing run in the backend.
 
 """
+import re
+import ast
+
 import numpy as np
-import pandas as pd
 import xarray as xr
 from calliope.core.attrdict import AttrDict
-from calliope.core.util.observed_dict import UpdateObserverDict
+from calliope.backend.subsets import imask_where, _combine_imasks, _imask_foreach
+
+
+def _equation_parser(model_data, eq_dict):
```

Comment:
Should this already go somewhere else assuming it will be used for custom constraints too?